include/engine/api/match_parameters.hpp

```diff
@@ -61,12 +62,16 @@ struct MatchParameters : public RouteParameters
     }
 
     template <typename... Args>
-    MatchParameters(std::vector<unsigned> timestamps_, Args... args_)
-        : RouteParameters{std::forward<Args>(args_)...}, timestamps{std::move(timestamps_)}
+    MatchParameters(
+        std::vector<unsigned> timestamps_,
+        const boost::optional<bool> allow_uturn_,
+        Args... args_)
+        : RouteParameters{std::forward<Args>(args_)...}, timestamps{std::move(timestamps_), allow_uturn{allow_uturn_}}
     {
     }
 
     std::vector<unsigned> timestamps;
+    boost::optional<bool> allow_uturn = true;
```

Comment:
Should this be an optional or a boolean with true being the default? Why the optional here?